cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -408,6 +412,70 @@ func (pm *PipelineManager) Configure(ctx context.Context, projectName string) (r
 		pm.configOptions.projectVariables, pm.configOptions.projectSecrets,
 		defaultAzdVariables, defaultAzdSecrets, pm.env.Dotenv())
 
+	// resolve akvs secrets
+	// For each akvs in the secrets array:
+	// azd gets the value from Azure Key Vault and use it as a secret in the pipeline
+	for key, value := range pm.configOptions.secrets {
+		if !strings.HasPrefix(value, "akvs://") {
+			continue
+		}
+		kvSecret, err := pm.keyVaultService.SecretFromAkvs(ctx, value)
+		if err != nil {
+			return result, fmt.Errorf("failed to resolve akvs '%s': %w", key, err)
+		}
+		pm.configOptions.secrets[key] = kvSecret
+	}
+	// For each akvs in the variables array:
+	// azd must grant read access role to the pipelines's identity to read the akvs
+	displayMsg = "Assigning read access role for Key Vault to service principal"
+	pm.console.ShowSpinner(ctx, displayMsg, input.Step)
+	kvAccounts := make(map[string]struct{})
+	for key, value := range pm.configOptions.variables {
+		if !strings.HasPrefix(value, "akvs://") {
+			continue
+		}
+
+		akvs, err := keyvault.ParseAkvs(value)
+		if err != nil {
+			return result, fmt.Errorf("failed to parse akvs '%s': %w", key, err)
+		}
+		kvId := akvs.SubscriptionId + akvs.VaultName
+		if _, ok := kvAccounts[kvId]; ok {
+			// skip if already assigned role for this key vault
+			continue
+		}
+
+		// can't use keyvaultService.Get() because it requires the resource group name and we don't save it for akvs
+		allKvFromSub, err := pm.keyVaultService.ListSubscriptionVaults(ctx, akvs.SubscriptionId)
+		if err != nil {
+			return result, fmt.Errorf(
+				"assigning read access role for Key Vault to service principal: %w", err)
+		}
+		var vaultResourceId string
+		foundKeyVault := slices.ContainsFunc(allKvFromSub, func(kv keyvault.Vault) bool {
+			if kv.Name == akvs.VaultName {
+				vaultResourceId = kv.Id
+				return true
+			}
+			return false
+		})
+		if !foundKeyVault {
+			return result, fmt.Errorf(
+				"assigning read access role for Key Vault to service principal: "+
+					"key vault '%s' not found in subscription '%s'", akvs.VaultName, akvs.SubscriptionId)
+		}
+
+		err = pm.entraIdService.CreateRbac(
+			ctx, akvs.SubscriptionId, vaultResourceId, keyvault.RoleIdKeyVaultSecretsUser, *servicePrincipal.Id)
+		if err != nil {
+			return result, fmt.Errorf(
+				"assigning read access role for Key Vault to service principal: %w", err)
+		}
```

Comment:
What happens when the user runs `azd pipeline config` multiple times? It doesn't look like we are checking to see if an RBAC role assignment was previously made and thinking that this could fail or potentially create duplicate assignments.

Reply:
Good to know - i was curious since our implementation uses `roleAssignmentId := uuid.New().String()` to generate a unique assignment id on each call.